src/index.ts

```diff
@@ -40,6 +52,40 @@ const createGenerator: CodegenGeneratorConstructor = (config, context) => {
 			return !!value.mediaType.mimeType.match('\\bjson$')
 		})
 
+		hbs.registerHelper('hasOperationSupportingMultipart', function(value: CodegenOperation | CodegenOperation[]): boolean {
+			if (isCodegenOperation(value)) {
+				return containsMultipartOperation([value])
+			} else {
+				return containsMultipartOperation(value)
+			}
+		})
+
+		hbs.registerHelper('toFileUploadProperties', function(properties: CodegenProperties): FileUploadProperty[] {
+			const results: FileUploadProperty[] = []
+			for (const prop in properties) {
+				const property = properties[prop]
+				if (property.schema.purpose !== CodegenSchemaPurpose.METADATA) {
+					continue
+				}
+
+				if (isCodegenArraySchema(property.schema) && property.schema.component) {
+					results.push({
+						name: prop,
+						minCount: property.schema.minItems,
+						maxCount: property.schema.maxItems,
+					})
+				} else {
+					results.push({
+						name: prop,
+						minCount: null,
+						maxCount: null,
```

Comment:
Should this be a `1` to validate that they don't submit two when we expect one?

Reply:
The 1 gets added in the `api.hbs` if `maxCount` is `null`, but it's probably a safer bet to do it here.